### ISSUE TITLE
There is a way to left a button in overlay to close the webview?

### ISSUE THREAD

**Issue by ikaro88** [gree/unity-webview#471]
There is a way to left a button in overlay to close the webview?

Now when i open the webview the only way to turn to my app is press back button as meny time as the pages i opened, There is a way to left a button in overlay to close the webview?

**Comment by KojiNakamaru** [comment]
You cannot overlay any unity-native UI element, as a native webview is placed over unity's rendering view. You need to place a button not overlapping webivews or to "inject" a html button as below. The latter approach is tricky so that it is better to preliminarily place a button in html if you can edit the html.

```diff
diff --git a/sample/Assets/Scripts/SampleWebView.cs b/sample/Assets/Scripts/SampleWebView.cs
index 1b7596e..d775682 100644
--- a/sample/Assets/Scripts/SampleWebView.cs
+++ b/sample/Assets/Scripts/SampleWebView.cs
@@ -33,6 +33,9 @@ public class SampleWebView : MonoBehaviour
         webViewObject.Init(
             cb: (msg) =>
             {
+                if (msg == "close") {
+                    Destroy(webViewObject.gameObject);
+                }
                 Debug.Log(string.Format("CallFromJS[{0}]", msg));
                 status.text = msg;
                 status.GetComponent<Animation>().Play();
@@ -93,6 +96,20 @@ public class SampleWebView : MonoBehaviour
 #endif
 #endif
                 webViewObject.EvaluateJS(@"Unity.call('ua=' + navigator.userAgent)");
+                webViewObject.EvaluateJS(
+                    "(function() {" +
+                    "  var button;" +
+                    "  button = document.createElement('button');" +
+                    "  button.style.position = 'fixed';" +
+                    "  button.style.right = '0px';" +
+                    "  button.style.top = '0px';" +
+                    "  button.style.zIndex = 2147483647;" +
+                    "  button.textContent = 'X';" +
+                    "  button.onclick = function() {" +
+                    "    Unity.call('close');" +
+                    "  };" +
+                    "  document.body.appendChild(button);" +
+                    "})()");
             },
             //ua: "custom user agent string",
             enableWKWebView: true);
```

**Comment by AndrewSpalato-FGMNT** [comment]
Doesn't seem to work on iOS, works on Android though but I have to include both snippets in the cb: and the started: parameters.

**Comment by KojiNakamaru** [comment]
It works for both platforms. `started:` is not adequate as the dom is not loaded yet when it is called. The followings are screenshots for Vankyo S7 (Android 9) and iPhone SE 2nd (iOS 14.4.2). My Unity version is 2019.4.18f1.
![image](https://user-images.githubusercontent.com/2645978/113401954-3ae4df00-93df-11eb-9a39-dc61bead6eef.png)
![image](https://user-images.githubusercontent.com/2645978/113401971-433d1a00-93df-11eb-8344-736f15d42fc1.png)

**Comment by AndrewSpalato-FGMNT** [comment]
@KojiNakamaru I can add a close button to an Android WebView with WkWebView, but I am unable to have the close button displayed on iOS. Is there anything different I must do in order for it to appear?

**Comment by KojiNakamaru** [comment]
There should be no extra step. Did you try the sample app with the modification above? It should work as in the screenshots. Although I don't have iPhone X, here is the simulator output for iPhone 12:

![image](https://user-images.githubusercontent.com/2645978/113662302-ca8dd480-96e2-11eb-8b92-21ec2102e41d.png)

**Comment by AndrewSpalato-FGMNT** [comment]
I eventually managed to get the close button visible, but it is not interactable.

**Comment by KojiNakamaru** [comment]
Again, did you try the sample app with the above modification? I guess you discarded the following part: https://github.com/gree/unity-webview/blob/3a537ee42a19bdff6c3983eec585fc37bb642a02/sample/Assets/Scripts/SampleWebView.cs#L61-L101

**Comment by AndrewSpalato-FGMNT** [comment]
Yeah the demo scene is all good, I am now trying to implement this on a specific url. 

It could be the Web AR content disallowing other input. Need to investigate further.

**Comment by OmarINNOVI** [comment]
Hello, im trying to Destroy the WebViewObject itself ( trying to close the webview by doing a certain action ) visually it seems destroyed yet i cannot interact anymore with UI elements, im i missing something ?

**Comment by KojiNakamaru** [comment]
@OmarINNOVI The sample app shows the "*" button that destroys/instantiates a WebViewObject, and the button seems to work okay. Can you reproduce your issue based on the sample app with your minimum modification?
